### ISSUE TITLE
CICP not working with SVT

### ISSUE THREAD

**Issue by cdgriffith** [AOMediaCodec/libavif#870]
Using an HDR10 PQ input with CICP 9/16/9 to try and convert to AVIF. 

AOM and rav1e both work as expected, but SVT-AV1 produces a bad result (muted colors, due to bad color transfer). I tried to manually specify the transfers as well and that did not work either. 

Working AOM command: 
```
ffmpeg -i .\artist.mp4 -to 1 -pix_fmt yuv420p10le -map 0:v:0 -f yuv4mpegpipe -strict -1 -r 30 - | avifenc --stdin --fps 30 --cicp 9/16/9 --speed 6 -c aom -j 16 aom_art_cicp.avif
```

SVT command with bad result: 
```
ffmpeg -i .\artist.mp4 -to 1 -pix_fmt yuv420p10le -map 0:v:0 -f yuv4mpegpipe -strict -1 -r 30 - | avifenc --stdin --fps 30 --cicp 9/16/9 --speed 6 -c svt -j 16 svt_art_cicp_no_advanced.avif
```

SVT tried with manual advanced key values:
```
ffmpeg -i .\artist.mp4 -to 1 -pix_fmt yuv420p10le -map 0:v:0 -f yuv4mpegpipe -strict -1 -r 30 - | avifenc --stdin --fps 30 --cicp 9/16/9 --speed 6 -c svt -j 16 -a color-primaries=9 -a transfer-characteristics=16 -a matrix-coefficients=9 svt_art_cicp.avif
```

The command output does look clean and says it will add the CICP information: 
```
AVIF to be written: (Lossy)
 * Resolution     : 1920x1080
 * Bit Depth      : 10
 * Format         : YUV420
 * Chroma Sam. Pos: 0
 * Alpha          : Absent
 * Range          : Limited
 * Color Primaries: 9
 * Transfer Char. : 16
 * Matrix Coeffs. : 9
 * ICC Profile    : Absent (0 bytes)
 * XMP Metadata   : Absent (0 bytes)
 * EXIF Metadata  : Absent (0 bytes)
 * Transformations: None
 * Progressive    : Unavailable
Encoding with AV1 codec 'svt' speed [6], color QP [24 (Medium) <-> 26 (Medium)], alpha QP [0 (Lossless) <-> 0 (Lossless)], tileRowsLog2 [0], tileColsLog2 [0], 1 worker thread(s), please wait...
 * Encoding frame 1 [1/30 ts]: (stdin)
Svt[info]: -------------------------------------------
Svt[info]: SVT [version]:       SVT-AV1 Encoder Lib v0.9.1
Svt[info]: SVT [build]  :       Visual Studio 2019 64 bit
Svt[info]: LIB Build date: Mar  8 2022 19:17:16
Svt[info]: -------------------------------------------
Svt[info]: Number of logical cores available: 1
Svt[info]: Number of PPCS 37
Svt[info]: [asm level on system : up to avx2]
Svt[info]: [asm level selected : up to avx2]
Svt[info]: -------------------------------------------
Svt[info]: SVT [config]: Main Profile   Tier (auto)     Level (auto)
Svt[info]: SVT [config]: Preset                                                         : 6
Svt[info]: SVT [config]: EncoderBitDepth / EncoderColorFormat / CompressedTenBitFormat  : 10 / 1 / 0
Svt[info]: SVT [config]: SourceWidth / SourceHeight                                     : 1920 / 1080
Svt[info]: SVT [config]: Fps_Numerator / Fps_Denominator / Gop Size / IntraRefreshType  : 60000 / 1000 / 129 / 2
Svt[info]: SVT [config]: HierarchicalLevels  / PredStructure                            : 4 / 2
Svt[info]: SVT [config]: BRC Mode / Rate Factor / SceneChange                           : CRF / 26 / 0
Svt[info]: -------------------------------------------
```

Attaching example output images [avif_example_images_issue.zip](https://github.com/AOMediaCodec/libavif/files/8210174/avif_example_images_issue.zip)   (Have to be played with mpv to see the proper colors. They do not work on Chrome.) 

Version: 0.9.3 (dav1d [dec]:0.9.2-0-g7b433e0, aom [enc/dec]:3.3.0, rav1e [enc]:0.5.1 (v0.5.1), svt [enc]:v0.9.1)  - from appveyor

**Comment by joedrago** [comment]
You're right, it does not appear that the CICP values are being passed to SVT inside of `codec_svt.c`.

**Comment by cdgriffith** [comment]
I don't use chrome to view them, just `mpv` and it plays both rav1e and aom fine, just the `svt` one has muted colors. 

Could be `mpv` thing? 

![color_compare](https://user-images.githubusercontent.com/3275435/157342871-8cd68a91-b33e-4738-ac29-acc454e89daa.png)

**Comment by joedrago** [comment]
Oh, I'm not familiar with MPV, but my guess here is MPV isn't treating it like an AVIF (reading items from the container, etc), it is simply reading it as a regular MP4 video file (in which AVIF sequences are technically legal video files with no audio track). In this path, MPV must be relying on the underlying AV1 payload's CICP to determine its color profile.

(scrolls up) You said this though:

> They do not work on Chrome

What does this mean? The files can't be opened at all on Chrome, or the AOM vs SVT files are interpreted _differently_ in Chrome?

This behavior is definitely due to our SVT implementation not passing CICP to the AV1 encoder (I'm unfamiliar with the SVT impl and how SVT exposes this), but I wouldn't expect Chrome to interpret these files any differently from one another.

**Comment by cdgriffith** [comment]
They open and play correctly, but colors are all screwed up. On my 10-bit (non HDR) monitor they are muted, whereas they appear blindingly over-saturated on my 8-bit monitor. 

![monitor compare](https://user-images.githubusercontent.com/3275435/157344698-2696acb5-6d2f-496c-aa1d-d43e9859aa7c.png)

**Comment by joedrago** [comment]
Wait, that is the AOM file on both sides. Do none of them display correctly in Chrome, in your mind? On the same setup, does MPV display the image in a way you think is correct?

**Comment by cdgriffith** [comment]
Correct, none of them play colors correctly on Chrome. (Hence why I was saying don't use it for a comparison. I assumed this issue affected everyone.)

The aom and rav1e ones play correctly in mpv (look same as source video). Neither SVT one plays correct colors in mpv.

**Comment by joedrago** [comment]
Chrome should have proper AVIF HDR support, and your AOM-encoded image looks reasonable to me in both Chrome and Vantage on my Mac (using the Pro Display XDR). It is near impossible to "show you" using an SDR image capture on GitHub of what I am seeing in HDR on this monitor, but inspecting the pixel values of your image, it corresponds to my expectations. The plate is ~225nits, the paper towel in the bottom left gets up to the low 300s here and there. With a 300 nits threshold, very little of the image is over that:

![image](https://user-images.githubusercontent.com/332342/157349624-8d7f54ea-f06d-4d3e-b27b-696c439eb7d6.png)

I can't speak to why it would appear wrong on your Windows machine. May I ask what your SDR slider is set to in your Windows display settings? Do you think it is wrong because a specific white in your image doesn't correspond to the white of the web page or something, or does it simply look "too bright"?

**Comment by joedrago** [comment]
Here's a tiny (SDR) clip:

https://user-images.githubusercontent.com/332342/157350180-b50ff2f8-6113-4779-9617-9dc084d0b8d3.mp4

(Yes, I realize how ridiculous, anecdotal, and not too helpful it is show an SDR recording of an HDR display 😄, I just wanted to try to give some kind of understanding of relative brightness I'm seeing with my setup over here. )

**Comment by cdgriffith** [comment]
Overall way too bright. Here is similar capture from chrome 98 on my andriod 12 phone (Samsung 21 ultra) with same issue. 
![Screenshot_20220308-185145_Chrome.jpg](https://user-images.githubusercontent.com/3275435/157351778-18da86c9-47f6-4e5e-909a-1552d7ab029d.jpg) Easiest to see that the hand is much brighter than it should be (using aom version again).

**Comment by cdgriffith** [comment]
I imagine the issue with chrome is that it checks for a 10-bit display, and if found will not apply a tonemap even if the monitor is SDR (this will only be a problem on monitors like mine that is 10-bit but not HDR.) 

Whereas it's probably over saturated to my eyes because VLC / mpv probably use a more "cinamatic" tonemaping that is darker like `hable`, vs Chrome which might use something brighter like `mobius`. 

Sorry to derail conversation away from main issue, hopefully passing through the cicp info in `codec_svt.c` fixes the issue with players like `mpv` 😄
